### DISCUSSION TITLE
How to use callbacks with multiple elements / instances

### DISCUSSION THREAD

**Discussion by sipitai** [KingSora/OverlayScrollbars#461] [Q&A]
If I have a single element, and want to get the scroll information from within the onScroll callback, I can use the following code:

```
const instance = $('.scrollbar_element').overlayScrollbars({
    callbacks: {
        onScroll: function(e) {
            const scroll_info = instance.scroll();
            const max_y = scroll_info.max.y;
            const pos_y = scroll_info.position.y;
            // do something
        }
    }
}).overlayScrollbars();
```
But if I have multiple elements, the same code cannot be used.

With multiple elements, `instance` requires an index, but there's no index to use.

Firstly, is this the correct way of getting scroll information from within a callback? I couldn't find any relevant documentation for this.

And secondly, if it is, when there are multiple elements, how do I get the scroll information for the correct element?

Also, in the same situation, how would I add a class to the correct element(s) from within a callback?

**Comment by KingSora** [comment]
@sipitai in `v1` the callbacks receive the instance which called them as the `this` context. So you could do something like:
```js
const instance = $('.scrollbar_element').overlayScrollbars({
    callbacks: {
        onScroll: function(e) {
            const scroll_info = this.scroll(); // use this as the instance
            const max_y = scroll_info.max.y;
            const pos_y = scroll_info.position.y;
            // do something
        }
    }
}).overlayScrollbars();
```

In `v2`, which I recommend updating to, the first argument of every event is the instance which the event was called from

**Reply by KingSora** [reply]
@sipitai did my answer resolve your question?
